src/main/java/com/suse/saltstack/netapi/client/SaltStackClient.java

```diff
@@ -97,14 +97,18 @@ public Token login(String username, String password)
      * @return authentication token as {@link Token}
      * @throws SaltStackException if anything goes wrong
      */
-    public Token login(String username, String password, String eauth)
+    public Token login(final String username, final String password, final String eauth)
             throws SaltStackException {
-        JsonObject json = new JsonObject();
-        json.addProperty("username", username);
-        json.addProperty("password", password);
-        json.addProperty("eauth", eauth);
+        Map<String, String> kwargs = new LinkedHashMap() {
```

Comment:
That's right @hustodemon, at least that was also my understanding of it. I would personally be fine with declaring parameters as `final` in order to ensure that these are not reassigned or changed inside that method. The code in #28 could even be simplified by that, or not? In there we explicitly create final versions of the input parameters for the async methods (`final String constUsername = username;`), which could be skipped if we declared them as final parameters.